search/node-search.go

```diff
@@ -26,8 +28,31 @@ type keyValue struct {
 	value *yaml.Node
 }
 
+type Seeking int
+
+// Indicates to search what component of key-value pairing findcomponent should return
+const (
+	KEY = iota
+	VALUE
+)
+
+// Parses filePath and attempts to create Node Object
+func MakeNode(filePath string) (*yaml.Node, error) {
+	var node yaml.Node
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+	marshErr := yaml.Unmarshal(data, &node)
+	return &node, marshErr
+}
+
+func FindKey(node *yaml.Node, path ...string) (*yaml.Node, error) {
```

Comment:
Is the return always either the key, or an error. Is the key always a string, or int for sequence? And would the key always be the last part of the path if it exists?